### ISSUE TITLE
How to dynamically update value of rate limit in createExpressMiddleware

### ISSUE THREAD

**Issue by dilshans2k** [spinlud/redis-sliding-rate-limiter#2]
Hi,
I am trying to dynamically update `limit` of rate-limiter. I am storing the value of limit in redis and have subscribed using `psubscribe` command of redis. So whenever the value at that key changes in redis, i want to update limit in RateLimiter too.
Is this even possible with current codespace?
Please help, Its urgent

**Comment by spinlud** [comment]
Yes, this is supported. `RateLimiter` exposes public setters for `limit`, `windowSize`, `windowUnit` and `limitOverheadFraction`, and every call to `get()` reads the current values, so an assignment takes effect on the very next request with no reload:

```js
subscriber.psubscribe('ratelimit:limit:*');

subscriber.on('pmessage', (pattern, channel, message) => {
  const nextLimit = Number.parseInt(message, 10);
  if (Number.isInteger(nextLimit) && nextLimit > 0) {
    limiter.limit = nextLimit;
  }
});
```

Use a dedicated connection for the subscriber, since Redis does not allow a subscribed connection to run other commands. Note that the setter does not validate the value, so guard it as shown.

If instead you need a limit computed per request (for example from the request itself), the Express middleware has `overrideLimit` / `overrideLimitFn` for that.

This is now documented in the README under "Changing the limit at runtime" and shipped in v7.0.2. Closing, but feel free to reopen if something does not work for you.
